### DISCUSSION TITLE
code style: why  all require libraries must be localized ?

### DISCUSSION THREAD

**Discussion by bigfish02** [apache/apisix#8416] [Q&A]
In apisix [CODE_STYLE](https://github.com/apache/apisix/blob/master/CODE_STYLE.md#module):

```
local ngx = ngx
local require = require
```
why ngx and require also need be localized?
In my opinion, it is redundant.

**Comment by tzssangglass** [comment]
read more: https://doc.yonyoucloud.com/doc/wiki/project/openresty-best-practice/local.html
